src/SparkConduit.sol

```diff
@@ -294,6 +296,15 @@ contract SparkConduit is UpgradeableProxied, ISparkConduit, IInterestRateDataSou
         return _rayDiv(amount, IPool(pool).getReserveNormalizedIncome(asset));
     }
 
+    // Note: This function rounds up to the nearest share to prevent dust in conduit state.
+    function _convertToWithdrawShares(address asset, uint256 amount)
+        internal view returns (uint256)
+    {
+        uint256 normalizedIncome = IPool(pool).getReserveNormalizedIncome(asset);
+
+        return _rayDiv(amount, normalizedIncome) + (amount % normalizedIncome > 0 ? 1 : 0);
+    }
```

Comment:
I would change this to `_convertToSharesRoundUp` and use this:

```
function _divup(uint256 x, uint256 y) internal pure returns (uint256 z) {
    unchecked {
        z = x != 0 ? ((x - 1) / y) + 1 : 0;
    }
}
```

Also the comment is not quite correct. It's not about dust, but more about preventing user to steal amounts from other users (albeit a very small amount).

Reply:
Will use `x * 1e27` as a param

Reply:
Updated and added testing for both withdraw in specific audit case as well as harness testing